### ISSUE TITLE
Broken Pipe Bug Fix Found

### ISSUE THREAD

**Issue by sediavindivoh** [DarkPacks/SevTech-Ages#3430]
**Bug Report SevTech Server:**
A specific player online causes some other players to disconnect, as well as sometimes himself. (Though not everyone. Some players are immune to the disconnection.) The players will repeatedly receive a poor connection until they time out with a java connection closed by remote host error message.

**Expected Behavior**
Specific players logging in should not cause nearly all other players on the server to begin disconnecting, and freezing up the server.

**Steps to reproduce**
I can't give the exact steps that caused the player (now to be called Ken) to be affected in the first place, but I can give a huge amount of information that I've collected while exploring the problem.

1.) Ken was not causing the poor connection / disconnection problem for other players when the server first launched. After the server was in use for about a week or two is when the problem began occurring.
2.) Several players are "targetted" by Ken. When they were online they became increasingly laggy and could not interact with the world. Ken would also disconnect, especially when the "targetted" players were online.
3.) When Ken **logged out**, the log file would contain the following error message:

> [Netty Epoll Server IO #0/ERROR] [FML]: NetworkDispatcher exception
io.netty.channel.unix.Errors$NativeIoException: syscall:writev(..) failed: Broken pipe
        at io.netty.channel.unix.FileDescriptor.writeAddresses(..)(Unknown Source) ~[minecraft_server.1.12.2.jar:?]

The above error would change the IO port #0 #1 #2 #3 #4 #5 #6 #7 #8 , always matching the port that Ken connected to when joining the server.

4.) To further diagnose the problem, I used the same server files, but changed the world to world_2. Ken stopped causing the issue for the server on world_2. This indicated that somewhere in the saved world files the issue was occurring.

5.) I deleted as many files as I could find matching Ken's User ID, but they did not resolve the problem.
We then manually copied the world, deleting half the world's folders at a time and having Ken log in and log out, checking for "Broken Pipe" errors, which, mind you, do NOT always occur on Ken's first log out. Often times it would take 2 or even potentially MORE login/logout before the broken pipe message began appearing. We continued this process, checking which folders were causing the broken pipe and deleting the others until we'd narrowed it down to exactly ONE world folder! We deleted server/SevTechAges/world_sevtech_ages/**galacticraft** folder and found that the issue no longer occurred  ('confirmed' this by having ken log in and out about 6-7 tries)!!! This means that the galacticraft folder very likely contains the corrupted files that were causing the problem.

Having spent about 7 hours working on the problem. We've all logged out of the server, so we haven't been able to test and confirm that this has fixed the issue for sure, but preliminary results were good.

**More information:**
The lag created from this bug sometimes got so bad that it caused the entire linux system to freeze (I've never seen Linux OS freeze before). The compute did not respond to the shut down button, and I had to flip the switch on the back to shut it down. Even SSH requests would not respond.

**Client Information:**
Modpack Version: 3.0.8
Java Version: version 8 update 181
Launcher Used: twitch
Memory Allocated: 8Gb
Server/LAN/Single Player: LAN and Server (online players experienced the connection problems more heavily)
Optifine Installed:no

**World Information:**
SevTech version 3.0.8.
Everyone is well before the galactic age. This occurred first in age 2.
Additional content installed: 'mycommands' mod was added. (home/back/sethome/warp commands mod)

**TLDR:** Delete galacticraft folder when experiencing this error (if your players are not yet in the space age?) and it may resolve your issue!

Please get back to me and let me know if this assisted your server (users), or if this helped you fix the bug (admins). I'll help with any other information I have and will definitely be checking back here.

Thanks for your hard work on this great modpack ;)

[2018-10-05-2.log](https://github.com/DarkPacks/SevTech-Ages/files/2460527/2018-10-05-2.log)

**Comment by sediavindivoh** [comment]
Unfortunately, the error occurred again just a moment ago. I was online with a different person named Que. Que and I were online for about an hour before a third person joined. As soon as the third person joined, he got extreme lag, and so did Que. (But never me for some reason. I am lanned to the network). Every time the third person reconnected he was met with extreme lag again. Then Que logged out and the infamous Broken Pipe error displayed its ugly face. After Que was logged out, the third person and I could play just fine.

So, the error has now jumped to a 2nd person. I anticipate while this person (Que) is online, no one will be able to play properly except for me.
